sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/TopicSubscriptionWithRuleOperationsSample.java

```diff
@@ -0,0 +1,210 @@
+            administrationClient.getRule(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME, CORRELATION_FILTER_SUBSCRIPTION_RULE_NAME)
+                .setFilter(new CorrelationRuleFilter().setCorrelationId("important").setLabel("Red")));
+
+        // Get Rules on Subscription, called here only for one subscription as example
+        administrationClient.listRules(SERVICE_BUS_TOPIC_NAME, CORRELATION_FILTER_SUBSCRIPTION_NAME)
+            .forEach(ruleProperties -> {
+                System.out.println(String.format("GetRules:: SubscriptionName: %s, CorrelationFilter Name: %s, Rule: %s", CORRELATION_FILTER_SUBSCRIPTION_NAME, ruleProperties.getName(), ruleProperties.getFilter()));
+            });
+
+        // Send messages to Topic
+        sendMessagesAsync();
+
+        // Receive messages from 'allMessagesSubscriptionName'. Should receive all 9 messages
+        receiveMessagesAsync(ALL_MESSAGES_SUBSCRIPTION_NAME);
+
+        // Receive messages from 'sqlFilterOnlySubscriptionName'. Should receive all messages with Color = 'Red' i.e 3 messages
+        receiveMessagesAsync(SQL_FILTER_ONLY_SUBSCRIPTION_NAME);
+
+        // Receive messages from 'sqlFilterWithActionSubscriptionName'. Should receive all messages with Color = 'Blue'
+        // i.e 3 messages AND all messages should have color set to 'BlueProcessed'
+        receiveMessagesAsync(SQL_FILTER_WITH_ACTION_SUBSCRIPTION_NAME);
+
+        // Receive messages from 'correlationFilterSubscriptionName'. Should receive all messages  with Color = 'Red' and CorrelationId = "important"
+        // i.e 1 message
+        receiveMessagesAsync(CORRELATION_FILTER_SUBSCRIPTION_NAME);
+
+    }
+
+    /**
+     * Receive {@link ServiceBusReceivedMessage messages} by topic subscriptions from an Azure Service Bus Topic.
+     *
+     * @param subscriptionName Subscription Name.
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    static void receiveMessagesAsync(String subscriptionName) throws InterruptedException {
+        AtomicReference<ServiceBusReceiverAsyncClient> receiverAsyncClient = new AtomicReference<>();
+        AtomicReference<CountDownLatch> countdownLatch = new AtomicReference<>();
+        countdownLatch.set(new CountDownLatch(1));
+        receiverAsyncClient.set(new ServiceBusClientBuilder()
+            .connectionString(SERVICE_BUS_CONNECTION_STRING)
+            .receiver()
+            .topicName(SERVICE_BUS_TOPIC_NAME)
+            .subscriptionName(subscriptionName)
+            .receiveMode(ServiceBusReceiveMode.RECEIVE_AND_DELETE)
+            .buildAsyncClient());
+        System.out.println("==========================================================================");
+        System.out.println(String.format("%s :: Receiving Messages From Subscription: %s", OffsetDateTime.now(), subscriptionName));
+        AtomicLong receiveMessagesCount = new AtomicLong(0L);
```

Comment:
This use of AtomicReferences is bizarre, we can do something like:
```java
var receiver = new ServiceBusClientBuilder()
            .connectionString(SERVICE_BUS_CONNECTION_STRING)
            .receiver()
            .topicName(SERVICE_BUS_TOPIC_NAME)
            .subscriptionName(subscriptionName)
            .receiveMode(ServiceBusReceiveMode.RECEIVE_AND_DELETE)
            .buildAsyncClient();
try {
    // Taking messages for 10 seconds.
    var receiveOperation =  receiver.receiveMessages().take(Duration.ofSeconds(10)).flatMap(message -> {
        System.out.printf("Color Property = %s, CorrelationId = %s%n", color, correlation);
        return receiver.complete(message);
    });

    // Wait until the receive operation is complete.
    Mono.when(receiveOperation).block();
} finally {
    receiver.close();
}
```

Reply:
According to your comment, fixed within new version